src/librustc_trans/context.rs

```diff
@@ -86,17 +83,13 @@ pub struct SharedCrateContext<'a, 'tcx: 'a> {
     check_overflow: bool,
 
     use_dll_storage_attrs: bool,
-
-    translation_items: RefCell<FxHashSet<TransItem<'tcx>>>,
-    trait_cache: RefCell<DepTrackingMap<TraitSelectionCache<'tcx>>>,
-    project_cache: RefCell<DepTrackingMap<ProjectionCache<'tcx>>>,
 }
```

Comment:
This reminds me, the design of `SharedCrateContext` was a mistake IMO - its fields should be public and `CrateContext` should `Deref` to it so we get e.g. `ccx.tcx` and also no duplication of methods.